core/xml.js

```diff
@@ -86,6 +86,34 @@ Blockly.Xml.blockToDomWithXY = function(block, opt_noId) {
   return element;
 };
 
+/**
+ * Encode a variable field as XML.
+ * @param {!Blockly.Field} field The field to encode.
+ * @return {?Element} XML element, or null if the field did not need to be
+ *     serialized.
+ * @private
+ */
+Blockly.Xml.fieldToDomVariable_ = function(field) {
+  var id = field.getValue();
+  // The field had not been initialized fully before being serialized.
```

Comment:
Added to comment:
```
//This can happen if a block is created directly through a call to
// workspace.newBlock instead of from XML.
// The new block will be serialized for the first time when firing a block
// creation event.
```